src/extensions/otel_viewer_extension/OtelData.cs

```diff
@@ -0,0 +1,91 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using Newtonsoft.Json;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+using Newtonsoft.Json.Linq;
+using Google.Protobuf;
+using Azure.AI.Details.Common.CLI;
+
+namespace Azure.AI.Details.Common.CLI.Extensions.Otel
+{
+    public class OtelData
+    {
+        public static async Task<string> GetTrace(string requestId)
+        {
+            // Endpoint URL with the request ID
+            string url = $"https://spanretriever1-webapp.whitehill-dc9ec001.westus.azurecontainerapps.io/getspan/{requestId}";
+            using (HttpClient client = new HttpClient())
+            {
+                try
+                {
+                    // Send a GET request to the endpoint
+                    HttpResponseMessage response = await client.GetAsync(url);
+                    response.EnsureSuccessStatusCode();
+                    string responseBody = await response.Content.ReadAsStringAsync();
+
+                    // Convert JSON to a formatted string
+                    var parsedJson = JsonConvert.DeserializeObject(responseBody);
+                    string formattedJson = JsonConvert.SerializeObject(parsedJson, Formatting.Indented);
+
+                    return formattedJson;
+                }
+                catch (HttpRequestException httpEx)
+                {
+                    Console.WriteLine("HTTP Request failed: " + httpEx.Message);
+                    return null;
+                }
+            }
+
+        }
+        public static async Task WriteDataToFile(string data, string filePath)
+        {
+            try
+            {
+                //var data = await GetTrace(requestId);
+                using (StreamWriter writer = new StreamWriter(filePath))
+                {
+                    await writer.WriteAsync(data);
+                    Console.WriteLine($"Payload successfully written to {filePath}");
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred while writing to the file: {ex.Message}");
+            }
+
+        }
+        public static async Task ExportToDashboard(string data)
+        {
+            try
+            {
+                
+                // Check if the dashboard (OTLP endpoint) is running
+                if (!Dashboard.IsRunning())
+                {
+                    Console.WriteLine("Dashboard is not running. Please start the dashboard and try again.");
+                    return; 
+                }
+                
+                var settings = new JsonSerializerSettings();
+                settings.Converters.Add(new AnyValueConverter());
+                settings.Converters.Add(new IntArrayToByteStringConverter());
+
+                var request = JsonConvert.DeserializeObject<ExportTraceServiceRequest>(data, settings);
+        
+                var otlpEndpoint = "http://localhost:4317";
```

Comment:
will it always be port 4317?

Reply:
yes, the dashboard will always be listening on port 4317 given how it is currently launched:
`docker run --rm -it -p 18888:18888 -p 4317:18889 -d --name aspire-dashboard mcr.microsoft.com/dotnet/aspire-dashboard:8.0.0`

Reply:
is that the port that the docs/samples recommend? should we offer it on another port via an option? or is this probably fine?